tumblr-photo-video-ripper.py

```diff
@@ -195,18 +206,21 @@ def _download_media(self, site, medium_type, start):
             if response.status_code == 404:
                 print("Site %s does not exist" % site)
                 break
-
             try:
-                xml_cleaned = re.sub(u'[^\x20-\x7f]+',
-                                     u'', response.content.decode('utf-8'))
-                data = xmltodict.parse(xml_cleaned)
+                data = xmltodict.parse(response.text)
                 posts = data["tumblr"]["posts"]["post"]
                 for post in posts:
                     try:
                         # if post has photoset, walk into photoset for each photo
-                        photoset = post["photoset"]["photo"]
-                        for photo in photoset:
-                            self.queue.put((medium_type, photo, target_folder))
+                        if medium_type == "text":
+                            pattern = re.compile('.*?<img.*?data-orig-src="(.*?)"/>', re.S)
+                            pic_urls = re.findall(pattern, post["regular-body"])
+                            for pic_url in pic_urls:
+                                self.queue.put((medium_type, pic_url, target_folder))
```

Comment:
You're using `text` as medium type here. This is not correct.

Here `medium_type` should be `photo`.